huaweicloud/services/obs/resource_huaweicloud_obs_bucket_acl.go

```diff
@@ -0,0 +1,473 @@
+			"public_permission": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        permissionSchema(),
+				Optional:    true,
+				Description: `Specifies the public permission.`,
+			},
+			"log_delivery_user_permission": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        permissionSchema(),
+				Optional:    true,
+				Description: `Specifies the log delivery user permission.`,
+			},
+			"account_permissions": {
+				Type:        schema.TypeList,
+				Elem:        accountPermissionsSchema(),
+				Optional:    true,
+				Description: `Specifies the account permissions.`,
+			},
+		},
+	}
+}
+
+func permissionSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"access_to_bucket": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: `Specifies the access to bucket. Valid values are **READ** and **WRITE**.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"READ", "WRITE",
+					}, false),
+				},
+			},
+			"access_to_acl": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: `Specifies the access to acl. Valid values are **READ_ACP** and **WRITE_ACP**.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"READ_ACP", "WRITE_ACP",
+					}, false),
+				},
+			},
+		},
+	}
+	return &sc
+}
+
+func accountPermissionsSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"domain_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Specifies the domain id to authorize. The domain id cannot be the bucket owner, 
+and must be unique.`,
+			},
+			"access_to_bucket": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: `Specifies the access to bucket. Valid values are **READ** and **WRITE**.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"READ", "WRITE",
+					}, false),
+				},
+			},
+			"access_to_acl": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: `Specifies the access to acl. Valid values are **READ_ACP** and **WRITE_ACP**.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"READ_ACP", "WRITE_ACP",
+					}, false),
+				},
+			},
+		},
+	}
+	return &sc
+}
+
+func getPermissionTypesFromRawMap(rawMap map[string]interface{}) []obs.PermissionType {
+	var accesses []string
+	if accessArray, ok := rawMap["access_to_bucket"].([]interface{}); ok {
+		accesses = append(accesses, utils.ExpandToStringList(accessArray)...)
+	}
+	if accessArray, ok := rawMap["access_to_acl"].([]interface{}); ok {
+		accesses = append(accesses, utils.ExpandToStringList(accessArray)...)
+	}
+
+	var permissionTypes []obs.PermissionType
```

Comment:
```suggestion
	permissionTypes := make([]obs.PermissionType, len(accesses ))
```

please allocate memory in advance for arrays with known length.